### PR TITLE
Add AGS10 Sensor

### PR DESCRIPTION
**Pull request by mak-42**
## Description:

Documentation for AGS10 TVOC Sensor.

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#6070

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/sensor/ags10.rst [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        update_interval: 60s
+
+Configuration variables:
+------------------------
+
+- **tvoc** (**Required**): The information for the total Volatile Organic Compounds sensor.
+
+  - **name** (**Required**, string): The name for the tvoc sensor.
+  - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
+  - All other options from :ref:`Sensor <config-sensor>`.
```
Comment (jesserockz) [inline components/sensor/ags10.rst]: We don't duplicate config lines that are linked in sensor core docs
```suggestion
- **tvoc** (**Required**): The information for the total Volatile Organic Compounds sensor.
  All options from :ref:`Sensor <config-sensor>`.
```

File: components/sensor/ags10.rst [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    # Example configuration entry
+    sensor:
+      - platform: ags10
+        tvoc:
+          name: TVOC
+        address: 0x1A
+        update_interval: 60s
+
+Configuration variables:
```
Comment (jesserockz) [inline components/sensor/ags10.rst:36]: `address` and `resistance` sensors are missing from the documentation.

File: components/sensor/ags10.rst [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      scan: true
+      frequency: 15kHz
+
+    # Example configuration entry
+    sensor:
+      - platform: ags10
+        tvoc:
+          name: TVOC
+        address: 0x1A
+        update_interval: 60s
```
Comment (jesserockz) [inline components/sensor/ags10.rst]: Example YAML should be basic to allow the user to read the documentation and understand how to configure it
```suggestion
    # Example configuration entry
    sensor:
      - platform: ags10
        tvoc:
          name: TVOC
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: components/sensor/ags10.rst (hunk 1 above)
Comment (mak-42) [reply components/sensor/ags10.rst]: Done.

File: components/sensor/ags10.rst (hunk 2 above)
Comment (mak-42) [reply components/sensor/ags10.rst:36]: `version` and `resistance`. The `address` was already described.
Done.

File: components/sensor/ags10.rst (hunk 3 above)
Comment (mak-42) [reply components/sensor/ags10.rst]: Done

File: components/sensor/ags10.rst [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+------------------------
+
+- **tvoc** (**Required**): The information for the total Volatile Organic Compounds sensor.
+  All other options from :ref:`Sensor <config-sensor>`.
+- **address** (*Optional*, int): Manually specify the I²C address of
+  the sensor. Defaults to ``0x1A``.
+- **update_interval** (*Optional*, :ref:`config-time`): The interval to check the
+  sensor. Defaults to ``60s``.
+- **version** (*Optional*, int): The firmware version of the sensor.
+- **resistance** (*Optional*, int): The initial value of the sesor resistance.
```
Comment (jesserockz) [inline components/sensor/ags10.rst]:
```suggestion
- **resistance** (*Optional*, int): The initial value of the sensor resistance.
```

File: components/sensor/ags10.rst [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+------------------------
+
+- **tvoc** (**Required**): The information for the total Volatile Organic Compounds sensor.
+  All other options from :ref:`Sensor <config-sensor>`.
+- **address** (*Optional*, int): Manually specify the I²C address of
+  the sensor. Defaults to ``0x1A``.
+- **update_interval** (*Optional*, :ref:`config-time`): The interval to check the
+  sensor. Defaults to ``60s``.
+- **version** (*Optional*, int): The firmware version of the sensor.
+- **resistance** (*Optional*, int): The initial value of the sensor resistance.
```
Comment (jesserockz) [inline components/sensor/ags10.rst]:
```suggestion
- **version** (*Optional*): The firmware version of the sensor.
  All options from :ref:`Sensor <config-sensor>`.
- **resistance** (*Optional*): The initial value of the sensor resistance.
  All options from :ref:`Sensor <config-sensor>`.
```

File: components/sensor/ags10.rst [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    sensor:
+      - platform: ags10
+        tvoc:
+          name: TVOC
+
+Configuration variables:
+------------------------
+
+- **tvoc** (**Required**): The information for the total Volatile Organic Compounds sensor.
+  All other options from :ref:`Sensor <config-sensor>`.
```
Comment (jesserockz) [inline components/sensor/ags10.rst]:
```suggestion
  All options from :ref:`Sensor <config-sensor>`.
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: components/sensor/ags10.rst [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+    :image: ags10.jpg
+    :keywords: AGS10
+
+The ``ags10`` sensor platform VOC sensor allows you to use your ASAIR AGS10
+(`datasheet <http://www.aosong.com/userfiles/files/Datasheet%20AGS10.pdf>`__,
+`ASAIR`_ ) sensors with
+ESPHome. The :ref:`I²C Bus <i2c>` is
+required to be set up in your configuration for this sensor to work.
+
+.. warning::
```
Comment (jesserockz) [inline components/sensor/ags10.rst]: A warning is a bit scary for this
```suggestion
.. note::
```

**Review by jesserockz — APPROVED** [review]

**Pull request merged**